### ISSUE TITLE
Incorrect __init__.py script installed when not using catkin_make

### ISSUE THREAD

**Issue by jbrindza** [ros/genmsg#47]
If you compile and install a package that contains python source code and messages the `<package_name>/__init__.py` file from the devel space gets installed into the install space overwriting the previously installed `__init__.py` from the source.

You can recreate this problem by building dynamic_reconfigure with catkin_make_isolated. The `install_isolated/lib/python2.7/dist-packages/dynamic_reconfigure/__init__.py` file is the one from `devel_isolated/dynamic_reconfigure/lib/python2.7/dist-packages/dynamic_reconfigure/__init__.py`.

I believe it is due to the install regex rule for installing `__init__.py` files from subdirectories. I assume this rule is matching `<pkg_name>/./__init__.py`

```
 install(
 DIRECTORY ${CATKIN_DEVEL_PREFIX}/${@(l)_INSTALL_DIR}/@pkg_name
 DESTINATION ${@(l)_INSTALL_DIR}
 FILES_MATCHING
 REGEX "${CATKIN_DEVEL_PREFIX}/${@(l)_INSTALL_DIR}/@(pkg_name)/.+/__init__.pyc?$"
 )
```

Strangely, this does not happen with `catkin_make`, I don't understand why its different.

Same behaviour using cmake version 2.8.12.2 and 2.8.7

**Comment by dirk-thomas** [comment]
The problem should have been addressed some time ago with #38. Can you please mention with which genmsg version on what ROS distribution you are having the problem?

**Comment by jbrindza** [comment]
Sorry about that, looks like I was using an old version. Updating to the latest version fixed the issue. Thanks.
